Simplified/NYPLNetworkQueue.swift

```diff
@@ -1,41 +1,57 @@
 import Foundation
 import SQLite
 
+/**
+ The NetworkQueue is insantiated once on app startup and listens
+ for a valid network notification from a reachability class. It then
+ will retry any queued requests and purge them if necessary.
+ */
 final class NetworkQueue: NSObject {
-  
+
+  static let shared = NetworkQueue()
+
+  override init() {
+    super.init()
+    NotificationCenter.default.addObserver(forName: NSNotification.Name.NYPLReachabilityHostIsReachable,
+                                           object: nil,
+                                           queue: nil) { notification in self.retryQueue() }
+  }
+
   static let StatusCodes = [NSURLErrorTimedOut,
-                     NSURLErrorCannotFindHost,
-                     NSURLErrorCannotConnectToHost,
-                     NSURLErrorNetworkConnectionLost,
-                     NSURLErrorNotConnectedToInternet,
-                     NSURLErrorInternationalRoamingOff,
-                     NSURLErrorCallIsActive,
-                     NSURLErrorDataNotAllowed,
-                     NSURLErrorSecureConnectionFailed]
-  static let MaxRetriesInQueue = 5
-  
+                            NSURLErrorCannotFindHost,
+                            NSURLErrorCannotConnectToHost,
+                            NSURLErrorNetworkConnectionLost,
+                            NSURLErrorNotConnectedToInternet,
+                            NSURLErrorInternationalRoamingOff,
+                            NSURLErrorCallIsActive,
+                            NSURLErrorDataNotAllowed,
+                            NSURLErrorSecureConnectionFailed]
+  let MaxRetriesInQueue = 5
+
+  let serialQueue = DispatchQueue(label: "org.nypl.labs.SimplyE.networkQueue")
```

Comment:
This is too picky perhaps, but I'd recommend the following instead:

```swift
let serialQueue = DispatchQueue(Bundle.main.bundleIdentifier!
                                + "."
                                + String(describing: NetworkQueue.self))
```

This code will continue to work correctly even if the bundle ID changes and it will allow refactoring tools to handle changes to the class name.